### PR TITLE
Update the TPR parser for Gromacs 2022

### PR DESCRIPTION
**Pull request by jbarnoud**
This commit adds the necessary test files. These were built using
gromacs 2022-rc1. The commit also adds the tests corresponding to these
files.

Finally, this commit updates the TPR parser itself based on the changes
in `src/gromacs/fileio/tpxio.cpp` between gromacs 2021.5 and gromacs
2022-rc1.

Gromacs 2022-rc1 is available at https://manual.gromacs.org/documentation/2022-rc1/download.html
The changelog is available at https://manual.gromacs.org/documentation/2022-rc1/release-notes/index.html

PR Checklist
------------
 - [X] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - ~[ ] Issue raised/referenced?~

### CODE REVIEW

**Review by richardjgowers — APPROVED** [review]

File: package/CHANGELOG [hunk 1]
```diff
@@ -40,6 +40,7 @@ Enhancements
     option from setup.cfg (Issue #3428, PR #3429).
   * Added R/S Chirality support to RDKit parsed molecules
   * Added MDAnalysis.lib.distances.minimize_vectors
+  * Supports TPR files produced by Gromacs 2022 (PR #3514)
```
Comment (IAlibay) [inline package/CHANGELOG]: tsk tsk tsk entries should be newest first

**Review by IAlibay — APPROVED** [review]
thanks @jbarnoud 

quick question - there's been a couple of new mdp options recently (softcore stuff + constant acceleration), is there a need to update `do_inputrec` ? It doesn't look like it's being used by the parser code, but if it's "dead" code we probably should label it as such.

**Comment by jbarnoud** [comment]
> is there a need to update `do_inputrec` ?

Oh! Good call! I'll have a look.

File: package/CHANGELOG (hunk 1 above)
Comment (jbarnoud) [reply package/CHANGELOG]: You can tell I haven't pushed for a while :sweat_smile:

**Comment by jbarnoud** [comment]
> thanks @jbarnoud
> 
> quick question - there's been a couple need mdp options recently (softcore stuff + constant acceleration), is there a need to update `do_inputrec` ? It doesn't look like it's being used by the parser code, but if it's "dead" code we probably should label it as such.

It is dead code indeed. I created issue #3515 and will address it in a separate PR for scope.

**Comment by IAlibay** [comment]
Thanks @jbarnoud !

**Pull request merged**